CHANGELOG.md

```diff
@@ -1,3 +1,141 @@
+
+* Adds support for per-table option for columnar storage
+
+* Adds `rebalance_table_shards` to rebalance the shards across the nodes
+
+* Adds `citus_drain_node` to move all shards away from any node
+
+* Enables single-node Citus for production workloads
+
+* Adds support for local distributed/subquery joins
+
+* Enables foreign keys between reference tables and local tables
+  (`citus.enable_local_reference_table_foreign_keys`)
+
+* Adds support for co-located/recurring correlated subqueries
+
+* Adds `alter_distributed_table` and `alter_table_set_access_method` UDFs
+
+* Adds `alter_old_partitions_set_access_method` UDF to compress old partitions
+
+* Adds cascade option to `undistribute_table` UDF for foreign key connected
+  relations
+
+* Allows distributed table creation immediately after CREATE EXTENSION citus
+
+* Automatically adds coordinator to the metadata before the first distributed
+  table created
+
+* Introduces adaptive connection management for local nodes
+
+* Adds `public.citus_tables` view
+
+* Adds `pg_catalog.citus_shards`, `pg_catalog.citus_shards_on_worker` and
+  `pg_catalog.citus_shard_indexes_on_worker` views
+
+* Adds `pg_catalog.time_partitions` view for simple (time) partitions and their
+  access methods
+
+* Adds `remove_local_tables_from_metadata` UDF for local tables automatically
+  added to metadata when defining foreign keys with reference tables
+
+* Adds support for `CREATE INDEX` commands without index name on citus tables
+
+* Adds support for `CREATE TABLE ... USING table_access_method` statements
+
+* Adds support for co-located/recurring sublinks in the target list
+
+* Adds support for local execution in `INSERT..SELECT` commands that perform
+  re-partitioning
+
+* Propagates `ALTER TABLE .. SET LOGGED/UNLOGGED` statements
+
+* Propagates `ALTER INDEX .. SET STATISTICS` statements
+
+* Propagates `ALTER SCHEMA RENAME` statements
+
+* Propagates `CREATE STATISTICS` statements across workers.
+
+* Propagates `DROP STATISTICS` statements across the workers
+
+* Propagates all types of `ALTER STATISTICS` statements.
+
+* Improves performance in some subquery pushdown cases
+
+* Removes `citus.binary_master_copy_format` GUC
+
+* Removes `citus.expire_cached_shards` GUC
+
+* Removes `citus.large_table_shard_count` GUC
+
+* Removes `citus.max_assign_task_batch_size` GUC
+
+* Removes `citus.max_running_tasks_per_node` GUC
+
+* Removes `citus.max_task_string_size` GUC
+
+* Removes `citus.max_tracked_tasks_per_node` GUC
+
+* Removes `citus.sslmode` GUC
+
+* Removes `citus.task_tracker_delay` GUC
+
+* Removes the word 'master' from Citus UDFs
+
+* Removes unused UDF `mark_tables_colocated`
+
+* Removes `upgrade_to_reference_table` UDF
+
+* Renames 'master' to 'distributed' for columns of `citus_dist_stat_activity`
+
+* Renames 'master' to 'distributed' for columns of `citus_worker_stat_activity`
+
+* Avoids establishing new connections until the previous ones are finalized
```

Comment:
These two ` Avoids establishing ..` and ` Avoids re-using connections for in ..` are actually significant as well. Shall we move these to just before ` Adds support for `CREATE INDEX ..`

Reply:
makes sense

Reply:
b37a3e6683c95c997a08a95495ee212388a94bed